### ISSUE TITLE
Crons: Warn on mismatched schedule

### ISSUE THREAD

**Issue by evanpurkhiser** [getsentry/sentry#44387]
There is a scenario in the current implementation of Crons monitors where we probably should not be letting the user know "everything is okay"

Imagine a crontab monitor configured as `0 * * * *`. This cron monitor should run on the hour every hour.

<img width="427px" src="https://user-images.githubusercontent.com/1421724/217949906-c30738d9-d21b-4ddb-88a2-31e2b67efbc8.png" />

What is happening here?
* The first checkin happens at `12:00`. This is on time and successful
* The second run happens at `1:00`. This is on time and successful.
* There are now five additional checkins which are sent **before** the next expected checkin at `2:00`.

### Current world

Right now record these as simply "Okay". The monitor is successful and in the UI you can see all these additional checkins.

### Proposed world

If I have a monitor that runs once an hour, **it may be crucial for me that it ONLY runs once per hour**. This scenario is an error.

We should warn the user that they are **sending unexpected checkins**.

### Technical implementation proposal

Instead of recording each subsequent checkin that is before the next expected checkin, we can bucket these with the most recent checkin and simply record that they happened. We can also update the status of the monitor to be `extranious_checkins` or something like that.

This additionally helps us **reduce the number of rows** within the `monitorcheckin` table, which is something we are currently keeping an eye on, since we are currently receiving many extraneous checkins.

**Comment by wedamija** [comment]
How would the bucketing work in a world where these are events? Since events are immutable, we can't easily merge them.

**Comment by evanpurkhiser** [comment]
When these are events it is not as critical that we are recording so many and we can just record them with an "extra_checkin" status and compute the count of extranious ones with snuba, right?

**Comment by wedamija** [comment]
It's not *as* critical, but there's still a cost to storing/querying these. Since we're not charging by volume, it might make sense to still have some basic rate limiting in place.

**Comment by evanpurkhiser** [comment]
This is also related https://github.com/getsentry/sentry/issues/44390

**Comment by wedamija** [comment]
This is pretty old, not sure if we'll still do this, but maybe it makes sense for this to be configurable?

**Comment by gaprl** [comment]
Yes, it should be configurable. Additionally, we should have a new color display for check-ins in the timeline as suggested by https://github.com/getsentry/sentry/issues/72102.
